gnomad_qc/v4/subset.py

```diff
@@ -1,269 +1,471 @@
+            mtds = apply_variant_qc_annotations(mtds, config)
+
+        if config.pass_only:
+            logger.info("Filtering to variants that passed variant QC...")
+            mtds = filter_to_pass_only(mtds, config)
+
+        if config.output_vds:
+            if config.output_partitions:
+                logger.info(
+                    "Generating VDS with %d partitions...", config.output_partitions
+                )
+                temp_vds_path = new_temp_file("subset", "vds")
+                mtds = mtds.write(temp_vds_path)
+                mtds = hl.vds.read_vds(
+                    temp_vds_path, n_partitions=config.output_partitions
+                )
+
+            logger.info("Writing VDS subset...")
+            mtds.write(f"{config.output_path}/subset.vds", overwrite=config.overwrite)
+
+        if config.output_vcf:
+            logger.info("Formatting VCF info fields for export...")
+            mtds = format_vcf_info_fields(mtds)
+            mtds = mtds.naive_coalesce(
+                config.output_partitions
+                if config.output_partitions
+                else mtds.n_partitions()
+            )
+            if config.split_multi:
+                format_dict = copy.deepcopy(FORMAT_DICT)
+                print(format_dict)
+                format_dict.pop("RGQ")
+
+            logger.info("Exporting VCF subset...")
             hl.export_vcf(
-                mt,
-                f"{output_path}/subset.vcf.bgz",
-                metadata=header_dict,
+                mtds,
+                f"{config.output_path}/subset.vcf.bgz",
+                metadata={"format": FORMAT_DICT},
                 parallel="header_per_shard",
                 tabix=True,
             )
 
-    if args.vds:
-        vds.write(f"{output_path}/subset.vds", overwrite=args.overwrite)
+        if config.export_meta:
+            logger.info("Subsetting and exporting metadata...")
+            meta_ht = process_metadata_export(meta_ht, vds.variant_data, config)
+            meta_ht = meta_ht.checkpoint(
+                f"{config.output_path}/metadata.ht", overwrite=config.overwrite
+            )
+            meta_ht.export(f"{config.output_path}/metadata.tsv.bgz")
 
-    if args.export_meta:
-        logger.info("Exporting metadata")
-        meta_ht = meta_ht.semi_join(vds.variant_data.cols())
-        if args.keep_data_paths:
-            data_to_drop = {"ukb_meta"}
-        else:
-            data_to_drop = {"ukb_meta", "cram", "gvcf"}
+        logger.info("Subset operation completed successfully!")
 
-        meta_ht = meta_ht.annotate(
-            project_meta=meta_ht.project_meta.drop(*data_to_drop)
-        )
-        meta_ht = meta_ht.checkpoint(
-            f"{output_path}/metadata.ht", overwrite=args.overwrite
-        )
-        meta_ht.export(f"{output_path}/metadata.tsv.bgz")
+    finally:
+        logger.info("Copying log to logging bucket...")
+        hl.copy_log(f"{config.output_path}/v4_subset.log")
```

Comment:
```suggestion
        hl.copy_log(get_logging_path("v4_subset"))
```
or maybe remove this import if you prefer writing out to `config.output_path`

Reply:
I'll remove the import. `get_logging_path` uses gnomad-tmp and adds  a "exomes_4.1_qc"  to the path and I didnt feel this belonged there.